modules/gateways/Blockonomics/Blockonomics.php

```diff
@@ -234,14 +278,22 @@ public function getMargin() {
 			->where('setting', 'Margin')
 			->value('value');
 	}
+
 	/*
 	 * Convert fiat amount to BTC
 	 */
-	public function getBitcoinAmount($fiat_amount, $currency) {
+	public function getNewAmount($fiat_amount , $blockonomics_currency = 'btc') {
+		$currency = getCurrency(getClientsDetails()['user_id'])['code'];
```

Comment:
why do we need an additional call ? I think the caller function will have the order currency. Why not pass that too ?